### PR TITLE
Cart links to backend

### PR DESCRIPTION
**Pull request by shareasefrazier**

### CODE REVIEW

File: IT-Wizards-API/src/main/java/org/LaunchCode/IT_Wizards_API/services/CartService.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        Cart cart;
+        CartItem cartItem;
+       try {
+            cart = cartRepository.findByUserId(userId).orElseThrow(() -> new CartNotFoundException(userId));
+       } catch(Exception e) {
+           cart = cartRepository.save(new Cart(user));
+        }
+        Item item = itemRepository.findById(newItemId)
+                .orElseThrow(() -> new IllegalArgumentException("Item not found"));
+        //TODO: Look if cartItem exists & if so up quantity by 1
```
Comment (joshualjames) [inline IT-Wizards-API/src/main/java/org/LaunchCode/IT_Wizards_API/services/CartService.java:42]: This todo line can be removed now, right?

File: IT-Wizards-UI/src/components/CartContext.jsx [hunk 2]
```diff
@@ -1,10 +1,16 @@
 import { createContext, useEffect, useState } from 'react';
 import { getItems } from '../services/viewItemsService';
-import { useParams } from 'react-router-dom';
+import { useNavigate, useParams } from 'react-router-dom';
```
Comment (joshualjames) [inline IT-Wizards-UI/src/components/CartContext.jsx:3]: It looks like this useNavigate isn't being used and can be removed

**Review by joshualjames — APPROVED** [review]
Two minor things that I think can be removed either now or during the next commit, they aren't anything serious. Otherwise LGTM

File: IT-Wizards-API/src/main/java/org/LaunchCode/IT_Wizards_API/services/CartService.java (hunk 1 above)
Comment (shareasefrazier) [reply IT-Wizards-API/src/main/java/org/LaunchCode/IT_Wizards_API/services/CartService.java:42]: Yes, I forgot to clear that Todo, I will do so on my next commit along with useNavigate in CartContext.

**Pull request merged**